src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -217,6 +220,94 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     }
   }
 
+  override async fetchTokenInfo(contractId: string): Promise<IToken | undefined> {
+    const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
+    const apiUrl = ethActiveNetworkPredefinedSettings.value.tokenMiddlewareUrl;
+    try {
+      // Temporary solution for fetching the ERC-20 token info.
+      // TODO Replace with our own node API
+      return new EthplorerService(apiUrl).fetchTokenInfo(contractId);
+    } catch (error: any) {
+      Logger.write(error);
+      return undefined;
+    }
+  }
+
+  override async transferToken(
+    amount: number,
+    recipient: string,
+    contractId: string,
```

Comment:
```suggestion
    tokenContractId: AssetContractId,
```

Reply:
Currently `AssetContractId` is
```
export type AssetContractId =
  | Encoded.ContractAddress
  | typeof AE_CONTRACT_ID
  | typeof BTC_CONTRACT_ID
  | typeof ETH_CONTRACT_ID;
  ```
Do you think adding `| string` to a type `AssetContractId` is actually what we need?

Reply:
I think at the end we need to replace the whole list with `string`. For now if adding `string` to the list fixes your issue lets do it.